include/gz/math/Quaternion.hh

```diff
@@ -543,7 +543,7 @@ namespace gz::math
     /// \param[out] _angle CCW angle in radians.
     public: void AxisAngle(Vector3<T> &_axis, T &_angle) const
     {
-      T len = this->qx*this->qx + this->qy*this->qy + this->qz*this->qz;
+      T len = sqrt(this->qx*this->qx + this->qy*this->qy + this->qz*this->qz);
       if (equal<T>(len, static_cast<T>(0)))
```

Comment:
```suggestion
      T len2 = this->qx*this->qx + this->qy*this->qy + this->qz*this->qz;
      if (equal<T>(len2, static_cast<T>(0), 1e-12))
```

Your proposed solution makes sense. An alternative implementation is to rename the `len` variable and pass `_epsilon=1e-12` to the [math::equal function in Helpers.hh](https://github.com/gazebosim/gz-math/blob/gz-math7_7.5.0/include/gz/math/Helpers.hh#L399-L407), which can avoids some `sqrt` calls. I'm fine either way, just wanted to mention this to see which you prefer.

Reply:
Done: [a0dd4a8](https://github.com/gazebosim/gz-math/pull/620/commits/a0dd4a8a8c4cdc53a0b1ebe8d12558508481e842)